docs/diagrams/RequiredSequenceDiagram.puml

```diff
@@ -0,0 +1,90 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":GradPadParser" as GradPadParser LOGIC_COLOR
+participant "R:RequiredCommand" as RequiredCommand LOGIC_COLOR
+participant "Rs:RequiredCommandStorage" as RequiredCommandStorage LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("required")
+activate LogicManager
+
+LogicManager -> GradPadParser : parseCommand("required")
+activate GradPadParser
+
+create RequiredCommand
+GradPadParser -> RequiredCommand
+activate RequiredCommand
+
+RequiredCommand --> GradPadParser : R
+deactivate RequiredCommand
+
+GradPadParser --> LogicManager : R
+deactivate GradPadParser
+
+LogicManager -> RequiredCommand : execute()
+activate RequiredCommand
+
+RequiredCommand -> Model : getGradPad()
+activate Model
+
+Model --> RequiredCommand : GradPad
+
+RequiredCommand -> Model : getModuleList()
+
+Model --> RequiredCommand : currentModules
+deactivate Model
+
+RequiredCommand -> RequiredCommand : setStorage()
+activate RequiredCommand
+
+create RequiredCommandStorage
+RequiredCommand -> RequiredCommandStorage : Rs
```

Comment:
Constructor calls should be the class name not the object's name. I think it can also be omitted since it's evident in this case :D
```suggestion
RequiredCommand -> RequiredCommandStorage : RequiredCommandStorage()
```

Reply:
o dang, didnt know about this HAHAH changed accordingly!